.eslintrc.js

```diff
@@ -96,6 +96,28 @@ module.exports = {
     // Rules for prettier.
     "prettier/prettier": errorIfStrict, // Makes Prettier issues warnings rather than errors.
     "sort-destructure-keys/sort-destructure-keys": errorIfStrict,
+    "check-file/filename-naming-convention": [
+      errorIfStrict,
+      {
+        // GraphQL fragments, mutations and queries
+        "src/gql/fragments/**/*.graphql": "CAMEL_CASE",
+        "src/gql/(mutations,queries)/**/*.graphql": "KEBAB_CASE",
+        // Cypress
+        "cypress/integration/**/*.ts": "SNAKE_CASE",
+        // Scripts
+        "scripts/**/*.{js,ts}": "KEBAB_CASE",
+        // JS and TS with exceptions
+        "src/**/!(vite-env.d|custom-queries)*.{js,ts}": "CAMEL_CASE",
+        // All tsx with exceptions
+        "src/**/!(use|getFormSchema|index|test-utils|toast-decorator|schemaFields|getColumnsTemplate|githubPRLinkify|jiraLinkify)*.tsx":
```

Comment:
Should we hardcode all of these exceptions or can we update these to match the pattern we have established? 
e.g. jiraLinkify -> JiraLinkify.tsx